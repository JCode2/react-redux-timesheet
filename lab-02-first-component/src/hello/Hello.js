import React, {Component} from 'react';

class Hello extends Component{
  constructor(props){
    super(props);
    this.state = {greeting:'Howdy',}
  };
  render(){
      return(
          <div>
            <h1>{this.state.greeting}</h1>
            <h2>{this.props.friend}</h2>
            <p>Congrats you have created your first React Component...kinda</p>
          </div>
        );
      }
}

Hello.defaultProps = {
  friend: 'Partner!!',
};

export default Hello;
