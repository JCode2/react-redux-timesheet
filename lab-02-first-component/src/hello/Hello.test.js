import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import Hello from './Hello';

describe('Hello World', () =>{});

it('renders', ()=>{
  expect(shallow(<Hello />)).toHaveLength(1);
});

it('should render with default Components', () =>{
  const component = shallow(<Hello />);

  expect(component).toIncludeText('Howdy');
  expect(component).toIncludeText('Partner');
});

it('should render with our props', () =>{
  const component = shallow(<Hello friend="Fred"/>);

  expect(component).toIncludeText('Howdy');
  expect(component).not.toIncludeText('Partner');
  expect(component).toIncludeText('Fred');

});

it('should match the snapshot', () =>{
  const component = renderer.create(<Hello friend="Luke"/>);
  expect(component.toJSON()).toMatchSnapshot();
});
