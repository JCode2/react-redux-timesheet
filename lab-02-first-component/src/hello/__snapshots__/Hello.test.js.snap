// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div>
  <h1>
    Howdy
  </h1>
  <h2>
    Luke
  </h2>
  <p>
    Congrats you have created your first React Component...kinda
  </p>
</div>
`;
